### PR TITLE
feat: CLI: Allow setting Helm values literal (#3601)

### PR DESCRIPTION
**Pull request by sfynx**
While you could already set values files using the `--values` flag using the CLI with `argocd app create` and `argocd app set`, and you already could set individual parameters from an external source using `--helm-set`, it was not yet possible to manage the literal `.spec.source.helm.values` value in an Application without resorting to a complicated `argocd app patch` escaped parameter or by generating the entire application YAML manifest by yourself.

Therefore, this PR adds a `--values-literal-file` flag to the `argocd app create` and `argocd app set` commands, which accepts a local file name or URL to a values file, which will be read and included as a multiline string in the application manifest. This is different from the `--helm-set-file` flag which expects the file in the chart itself.

The `argocd app unset` command is expanded with a `--values-literal` flag, so we can also unset this field again.

I hope I chose nice enough names for the flags, I wanted to make clear it expects a file name, but also distinguish it enough from the existing `--values` flag which actually points to values files.

Because the current `setHelmOpt()` functionality would not work for unsetting things to an empty value and it was difficult to do these changes independently, this PR also contains the fix for issue #3644. A separate PR has still been created for that one because I think it should end up as a separate issue in the release notes.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by sfynx** [comment]
Not sure how I could reasonable cover more of the 24 lines that have been added, it is mainly added CLI structure.

**Review by jannfis — CHANGES_REQUESTED** [review]
Hi @sfynx, thanks a lot for your contribution!

Please ignore the Sonarcloud quality gate for now, we have no good mechanisms for unit testing the CLI yet, so we don't expect people to deliver unit tests for CLI changes. However, it would be great if you could write an end-to-end test (have a look at `test/e2e` directory) that ensures your feature works as expected.

**Comment by sfynx** [comment]
I have added the tests for both the local and remote file scenarios for adding a literal values block to an application, and also checked that it is cleared out as well when unsetting it.

**Review by jannfis — APPROVED** [review]
This is awesome, thank you! 

Changes look good to me - @alexmt, your say?

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -836,17 +856,20 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
...
-				specValueFiles := app.Spec.Source.Helm.ValueFiles
+				if valuesLiteral {
+					app.Spec.Source.Helm.Values = ""
+					updated = true
+				}
 				for _, valuesFile := range valuesFiles {
+					specValueFiles := app.Spec.Source.Helm.ValueFiles
 					for i, vf := range specValueFiles {
 						if vf == valuesFile {
-							specValueFiles = append(specValueFiles[0:i], specValueFiles[i+1:]...)
```
Comment (alexmt) [inline cmd/argocd/commands/app.go:843]: 👍

**Review by alexmt — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**